src/ParserContext.php

```diff
@@ -22,12 +30,46 @@ public function __construct() {
     $this->inlineContext = new InlineContext();
   }
 
+  public function setSourceType(SourceType $type): this {
+    switch ($type) {
+      case SourceType::TRUSTED:
+        $this->enableAllFeaturesForTrustedInput_UNSAFE();
+        break;
+      case SourceType::SPONSORED:
+        $this->setURISchemeAllowList();
```

Comment:
It would probably be better to remove the default parameter to force callsites to be more clear: I assumed this was setting the empty set.

This could be something like `(self::DEFAULT_URI_SCHEME_ALLOW_LIST)` ?

Reply:
This naming was also inconsistent with existing APIs, but this is what we should be using - please rebase on top of 76aa10c99def609c65f41fcf19d1f3802185bd37